frontend/src/static/js/api.js

```diff
@@ -93,4 +93,40 @@ window.api = {
         });
   },
 
+  calcCommitsVariance() {
+    const repos = window.REPOS;
+    const authorCommits = {};
+    const authorVariance = {};
+
+    Object.keys(repos).forEach((repo) => {
+      let repoCommits = 0;
+      let repoVariance = 0;
+      repos[repo].users.forEach((author) => {
+        repoCommits += author.totalCommits;
+        repoVariance += author.variance;
+        if (!Object.keys(authorCommits).includes(author)) {
+          authorCommits[author.name] = 0;
+          authorVariance[author.name] = 0;
+        }
+        // Accumulate author's commits and variance
+        authorCommits[author.name] += author.totalCommits;
```

Comment:
what prevents you from doing `author.authorCommits += author.totalCommits;` directly?

Reply:
If totalCommits/variance is updated directly within the loop, the values that are changed only affects that repo, but not other repos that the author may have.

After reimplementing upon your suggestion, the sorting within groups became inconsistent, hence the bug.

Reply:
I see.

I am still not quite convinced by the need to have 2 separate schemes in order to do the sorting of variance and commits, it simply implies that the existing sorting function is flawed.

Perhaps the target area to solve this problem shouldn't be using `repos`, but rather the sorting of filtered in the grouping function itself?

https://github.com/reposense/RepoSense/blob/011977ffb6291fe2ac3c2fc1996a409e9b4134b3/frontend/src/static/js/v_summary.js#L487-L541

Especially this particular line (below):

https://github.com/reposense/RepoSense/blob/011977ffb6291fe2ac3c2fc1996a409e9b4134b3/frontend/src/static/js/v_summary.js#L536

Reply:
It is possible to generate all total commits and variances from every author and repo, without `calcCommitsVariance()`. However this tabulating of values will be done every time the user sorts the ramp charts, inside the `groupBy...()` function.

Under situations where the repository is very big or there are many users, could there be a risk of slowing down the sorting process, thereby affecting user experience?

Reply:
The grouping and sorting is already done in the `groupBy` function anyway, I don't think even a 10000 sum operation can jeopardize the processing speed / operation.